src/postamble.js

```diff
@@ -44,8 +43,15 @@ if (memoryInitializer) {
   } else if (Module['memoryInitializerPrefixURL']) {
     memoryInitializer = Module['memoryInitializerPrefixURL'] + memoryInitializer;
   }
-  if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
-    var data = Module['readBinary'](memoryInitializer);
+  var data = Module['memoryInitializerBytes'];
+  if (!data) {
+    if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
+      data = Module['readBinary'](memoryInitializer);
+    } else {
+      data = tryParseAsDataURI(memoryInitializer);
```

Comment:
Well, this was a workaround for that issue where XHRing a data URI wouldn't be guaranteed to be successful because of Content Security Policy.

At this point, the fact that it's technically a data URI is more an implementation detail of the base64 embedding than anything else; `data:application/octet-stream;base64,` is being used as just an arbitrary prefix to tell all the relevant functions to decode the "file name" as base64 instead of pulling it from the network/filesystem, and could be changed to something else.